src/libs/actions/IOU.js

```diff
@@ -616,23 +667,33 @@ function createSplitsAndOnyxData(participants, currentUserLogin, amount, comment
                     },
                 },
             },
-        );
-
-        if (!existingOneOnOneChatReport) {
-            failureData.push({
+            ...(existingOneOnOneChatReport
+                ? []
+                : [
+                      {
+                          onyxMethod: Onyx.METHOD.MERGE,
+                          key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oneOnOneChatReport.reportID}`,
+                          value: {
+                              [oneOnOneCreatedActionForChat.reportActionID]: {
+                                  errors: {
+                                      [DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage'),
+                                  },
+                              },
+                          },
+                      },
+                  ]),
+            {
                 onyxMethod: Onyx.METHOD.MERGE,
-                key: `${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.reportID}`,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oneOnOneIOUReport.reportID}`,
                 value: {
-                    hasOutstandingIOU: existingOneOnOneChatReport ? existingOneOnOneChatReport.hasOutstandingIOU : false,
-                    iouReportID: existingOneOnOneChatReport ? existingOneOnOneChatReport.iouReportID : null,
-                    errorFields: {
-                        createChat: {
-                            [DateUtils.getMicroseconds()]: Localize.translateLocal('report.genericCreateReportFailureMessage'),
+                    [oneOnOneIOUAction.reportActionID]: {
```

Comment:
Shouldn't this be `oneOnOneCreatedActionForIOU` instead of `oneOnOneIOUAction`? (if not `existingOneOnOneIOUReport`)

Or maybe it's the other way around? https://github.com/Expensify/App/blob/85e7e5b9017566fcaa4819aec11935cb55045011/src/libs/actions/IOU.js#L217
here we should use `iouAction` instead of `iouCreatedAction` because I see this one is already missing the `isNewIOUReport` condition.


Either way let's be consist here about where the error message should be at.